libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -71,6 +71,8 @@ assembly::Statement Parser::parseStatement()
 		expectToken(Token::Colon);
 		assignment.variableName.location = location();
 		assignment.variableName.name = m_scanner->currentLiteral();
+		if (instructions().count(assignment.variableName.name))
+			fatalParserError("Identifier expected.");
```

Comment:
In the other the error will point at the `:=` symbol, so I would rather leave it as it is now.